src/mount.ts

```diff
@@ -20,16 +26,80 @@ export type MountOptions = {
   prepareContainer?: (container: HTMLElement) => void;
 };
 
+export type EnzymePropSelector = {
+  [key: string]: any;
+};
+
+export type ComponentType<Props> =
+  | ComponentClass<Props>
+  | FunctionComponent<Props>;
+
+export type EnzymeSelector = string | ComponentType<any> | EnzymePropSelector;
+
+/**
+ * Inspired by DefinitelyTyped's ReactWrapper.
+ * @see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cebb88fecfa52f854826e216a537867c11b0150e/types/enzyme/index.d.ts
+ */
+export type ComponentWrapper<P = {}, S = {}, C = Component> = {
+  at(index: number): ComponentWrapper;
+
+  childAt(index: number): ComponentWrapper<any, any>;
+  childAt<P2, S2>(index: number): ComponentWrapper<P2, S2>;
+
+  closest<P2>(
+    statelessComponent: FunctionComponent<P2>,
```

Comment:
The original types must have been written pre-hooks when function components were actually stateless. Now they are not if they use hooks.

Reply:
True. I removed the stateless variations.